src/utils.ts

```diff
@@ -0,0 +1,218 @@
+import {
+	FunctionDeclaration,
+	BlockStatement,
+	UnaryExpression,
+	CallExpression,
+	Node,
+} from "estree";
+import * as recast from "recast";
+
+export function isBooleanOperator(node: UnaryExpression) {
+	return (
+		node.type === "UnaryExpression" &&
+		node.operator === "!" &&
+		node.argument.type === "Literal" &&
+		typeof node.argument.value === "number"
+	);
+}
+
+export function isSpreadOperator(node: CallExpression): boolean {
+	return !!(
+		node.callee &&
+		(node.callee.type === "Identifier" ||
+			node.callee.type === "SequenceExpression" ||
+			node.callee.type === "MemberExpression") &&
+		node.arguments.length >= 2
+	);
+}
+
+///
+/// Spread operators end up like Y({}, props, {[key]: value})
+/// This is equivalent to:
+/// Object.assign({}, props, {[key]: value})
+/// So we want to turn it into:
+/// { ...props, [key]: value }
+
+export function parseSpreadOperator(node: CallExpression): any[] {
+	const returnNodes: any[] = [];
+
+	for (const argument of node.arguments) {
+		if (argument.type === "Identifier") {
+			returnNodes.push({
+				type: "JSXSpreadAttribute",
+				argument,
+			});
+			continue;
+		}
+
+		returnNodes.push(...parseComponentProps(argument, true));
+	}
+
+	return returnNodes;
+}
+
+export function parseComponentProps(originalProps: any, inSpread = false): any[] {
+	const attributes: any[] = [];
+
+	if (!originalProps) {
+		return [];
+	}
+
+	switch (originalProps.type) {
+		case "ObjectExpression": {
+			for (const prop of originalProps.properties) {
```

Comment:
Convert SpreadElements to JSXSpreadAttributes
```suggestion
			for (const prop of originalProps.properties) {
				if (prop.type === "SpreadElement") {
					attributes.push({
						type: "JSXSpreadAttribute",
						argument: prop.argument,
					});
					continue;
				}
```